libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/auditing/helpers/AuditExtensionsHelper.java

```diff
@@ -1,130 +1,122 @@
+        } else {
+            LOGGER.warn(LogHelper.buildLogMessage("VC not of type IdentityCheckCredential."));
+            return new AuditRestrictedF2F(null);
+        }
     }
 
     public static AuditRestrictedInheritedIdentity getRestrictedAuditDataForInheritedIdentity(
-            VerifiableCredential vc, String deviceInformation) throws CredentialParseException {
-        var credentialSubject =
-                OBJECT_MAPPER
-                        .valueToTree(vc.getClaimsSet())
-                        .path(CLAIMS_CLAIM)
-                        .path(VC_CLAIM)
-                        .path(VC_CREDENTIAL_SUBJECT);
-
-        try {
+            VerifiableCredential vc, String deviceInformation)
+            throws HttpResponseExceptionWithErrorBody {
+        if (vc.getCredential() instanceof IdentityCheckCredential identityCheckCredential) {
+            var credentialSubject = getCredentialSubjectOrThrow(identityCheckCredential);
+
             return new AuditRestrictedInheritedIdentity(
-                    OBJECT_MAPPER.treeToValue(credentialSubject.path(VC_NAME), LIST_NAME_TYPE),
-                    OBJECT_MAPPER.treeToValue(
-                            credentialSubject.path(VC_BIRTH_DATE), LIST_BIRTH_DATE_TYPE),
-                    OBJECT_MAPPER.treeToValue(
-                            credentialSubject.path(VC_SOCIAL_SECURITY_RECORD),
-                            LIST_SOCIAL_SECURITY_RECORD_TYPE),
+                    credentialSubject.getName(),
+                    credentialSubject.getBirthDate(),
+                    credentialSubject.getSocialSecurityRecord(),
                     new DeviceInformation(deviceInformation));
-        } catch (JsonProcessingException e) {
-            throw new CredentialParseException("Unable to parse VC for inherited ID audit data", e);
+        } else {
+            LOGGER.error(LogHelper.buildLogMessage("VC must be of type IdentityCheckCredential."));
+            throw new HttpResponseExceptionWithErrorBody(
+                    500, ErrorResponse.UNEXPECTED_CREDENTIAL_TYPE);
         }
     }
 
-    private static List<Name> getNameFromCredentialSubject(JsonNode credentialSubject)
-            throws CredentialParseException {
-        try {
-            return OBJECT_MAPPER.treeToValue(credentialSubject.path(VC_NAME), LIST_NAME_TYPE);
-        } catch (JsonProcessingException e) {
-            throw new CredentialParseException("Unable to get name list from credential subject");
+    private static IdentityCheckSubject getCredentialSubjectOrThrow(
+            IdentityCheckCredential credential) throws HttpResponseExceptionWithErrorBody {
+        var credentialSubject = credential.getCredentialSubject();
+
+        if (credentialSubject == null) {
+            LOGGER.error(
+                    LogHelper.buildLogMessage(
+                            ErrorResponse.CREDENTIAL_SUBJECT_MISSING.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
```

Comment:
Can we make this a warning instead? You might be able to modify some of your pattern matching conditions to make this easier to manage, e.g.:
```
if (vc.getCredential().getCredentialSubject() instanceof IdentityCheckSubject credentialSubject) {
```